controllers/auth_config_controller.go

```diff
@@ -608,7 +609,7 @@ func (r *AuthConfigReconciler) addToIndex(ctx context.Context, resourceNamespace
 
 	for _, host := range hosts {
 		// check for host name collision between resources
-		if indexedResourceId, found := r.Index.FindId(host); found && indexedResourceId != resourceId {
+		if indexedResourceId, found := r.Index.FindId(host); found && indexedResourceId != resourceId && (!r.AllowSupersedingHostSubsets || utils.SliceContains(r.Index.FindKeys(indexedResourceId), host)) {
```

Comment:
I'd maybe extract that one bool condition into a method of the reconciler... really a minor thing, but I find it makes it really hard to read when `&&` and `||` are mixed, and when there is a `!` in there just to screws with my brain ;)

Reply:
Done.